test/pylib/scylla_cluster.py

```diff
@@ -638,17 +639,28 @@ async def node_replace(self, old_node_id) -> Optional[str]:
 
 class Harness:
     """Manages a pool of Scylla clusters for running test cases"""
-    def __init__(self, test_name: str, clusters: Pool[ScyllaCluster]) -> None:
+    def __init__(self, test_name: str, base_dir: str, clusters: Pool[ScyllaCluster]) -> None:
         self.test_name: str = test_name
         self.clusters: Pool[ScyllaCluster] = clusters
         self.cluster: Optional[ScyllaCluster] = None  # Current cluster
         self.is_before_test_ok: bool = False
         self.is_after_test_ok: bool = False
         self.is_running: bool = False
+        # API
+        # NOTE: need to make a safe temp dir as tempfile can't make a safe temp sock name
```

Comment:
Clusters and their Pool exist outside a test suite. Harness "borrows" clusters and once it's done, if the cluster is not dirty, returns them to the Pool. Clusters have their own directory where their servers put files. Harness socket can't (and shouldn't) be inside any given cluster. And since there's no way to safe socket name in Python, the pattern is to create a safe directory and put the socket in there, instead.